packages/x-date-pickers/src/ClockPicker/ClockPicker.tsx

```diff
@@ -298,36 +301,23 @@ export const ClockPicker = React.forwardRef(function ClockPicker<TDate extends u
       switch (viewType) {
         case 'hours': {
           const value = convertValueToMeridiem(rawValue, meridiemMode, ampm);
-
-          if (date == null) {
-            return !isValidValue(value);
-          }
-
-          const dateWithNewHours = utils.setHours(date, value);
+          const dateWithNewHours = utils.setHours(dateOrMidnight, value);
```

Comment:
`dateOrMidnight` is the variable used on the `onChange` callbacks.
So using the same value should keep the validation coherent.

Even if `disableIgnoringDatePartForTimeValidation` is set to `true` (i.e: when used in `DateTimePicker` and there is a `minDateTime` or `maxDateTime` provided).